### ISSUE TITLE
Best way to fine tune on Wiki Data

### ISSUE THREAD

**Issue by JulianBvW** [Lightning-AI/lit-llama#365]
I want to fine-tune LLaMa on data I got from a fandom wiki ([for example this page](https://minecraft.fandom.com/wiki/Iron_Ingot)) and was wondering how to design the json file with its "prompt", "input", and "output"?

I can't just only use the prompt "Write the next sentance" and then put two adjacent sentences in input and output, right?

**Comment by rasbt** [comment]
One way would be to use the Dolly 2.0 JSON file as a template and structure your dataset in the same fashion, using the same keys. And then run the [prepare_dolly](https://github.com/Lightning-AI/lit-llama/blob/main/scripts/prepare_dolly.py) script.

**Comment by rasbt** [comment]
Or, maybe even easier would be to structure it similar to the Alpaca dataset, which has 
slightly different names for the keys, and then use the [prepare_alpaca](https://github.com/Lightning-AI/lit-llama/blob/main/scripts/prepare_alpaca.py) script.

<img width="691" alt="Screenshot 2023-06-05 at 10 52 34 AM" src="https://github.com/Lightning-AI/lit-llama/assets/5618407/1f4615dc-4bda-4e80-afae-f6cf08864bc3">

**Comment by JulianBvW** [comment]
Yes, but the question is how i can automatically fill instruction, input, and output using the web-scraped texts from the wiki pages?

**Comment by asadabbas09** [comment]
I'm also looking for the best way of creating dataset. I suppose we have to manually create some dataset (instructions/ output) manually and then can use self instruct to expand this and use for training. 

I'm not sure how much data do we need to create and what should be the length of each instruction, response. 

Is there a more systematic way of creating manual data?

**Comment by JuicyStandoffishMan** [comment]
I'm very new to this, so I apologize if it's incorrect, but I believe you can just follow the [unstructured data guide](https://github.com/Lightning-AI/lit-llama/blob/main/howto/unstructured_dataset.md) or adapt the [prepare_shakespeare.py](https://github.com/Lightning-AI/lit-llama/blob/main/scripts/prepare_shakespeare.py) code.
